### DISCUSSION TITLE
partialSafeParse data inference not working

### DISCUSSION THREAD

**Discussion by pfe-nazaries** [JacobWeisenburger/zod_utilz#16] [Q&A]
Hi, 

this code (similar to just plain safeParse) should infer the type of data if the validation is successful

```typescript
import { partialSafeParse } from "zod_utilz";
import { z } from "zod";

const schema = z.object({
    id: z.number()
})

const maybeValid = partialSafeParse(schema, { id: 1 });

if (maybeValid.success) {
    console.log(maybeValid.data.id);
}

```

Instead the type inferred is unknown. The same happens to the ZodErrors type

![image](https://github.com/user-attachments/assets/8f74dd34-e762-4a68-8274-23b8936bb657)

**Comment by JacobWeisenburger** [comment]
If you are only wanting to access the `data` property, then why are you using `partialSafeParse`? `zod` by itself works great for this.

```ts
const schema = z.object( { id: z.number() } )
const parsed = schema.safeParse( { id: 1 } )

if ( parsed.success ) {
    type Data = typeof parsed.data
    // type Data = { id: number }
}
```

---

`partialSafeParse` should only be used when you are wanting to extract only the `validData` from the result even if there is an invalid field.

```ts
const schema = z.object( { id: z.number(), name: z.string() } )
const parsed = zu.partialSafeParse( schema, { id: 1, name: false } )

console.log( parsed.validData ) // { id: 1 }
type ValidData = typeof parsed.validData
// type ValidData = {
//     name?: string | undefined
//     id?: number | undefined
// }

console.log( parsed.invalidData ) // { name: false }
type InvalidData = typeof parsed.invalidData
// type InvalidData = {
//     name?: string | undefined
//     id?: number | undefined
// }
```

**Comment by pfe-nazaries** [comment]
Hi, thanks for the quick answer. 

I use the best of both worlds, but for the repro example I just tried to show the problem in the simplest way possible

In my code what I do is a recovery system following these steps:

1. If validation is ok (.success === true), then return .data (inference not working)
2. If not return the valid data (.validData) and fill with default values the fields missing (invalidData) and the errors 

![image](https://github.com/user-attachments/assets/17cd4596-34af-4cff-a0b3-75dabb9c39ea)

My expectation with partialSafeParse is to have the same API of safeParse but with vitamins for partial data.

**Reply by JacobWeisenburger** [reply]
> My expectation with partialSafeParse is to have the same API of safeParse but with vitamins for partial data.

`safeParse` and `partialSafeParse` are not the same function and therefore they don't have the same API. They are not used for the same things.

**Comment by JacobWeisenburger** [comment]
`partialSafeParse` is literally already returning early if the normal zod safeParse works fine. so you are just duplicating that step

https://github.com/JacobWeisenburger/zod_utilz/blob/622cf8803d9cb7f734f4ec0d459fa4396f781332/src/partialSafeParse.ts#L31-L37
